project-base/storefront/components/Pages/Order/PaymentConfirmation/helpers.ts

```diff
@@ -77,11 +62,11 @@ export const usePaymentConfirmationContent = (
     };
 
     useEffect(() => {
-        if (!paymentStatusData) {
-            return;
+        if (!wasUpdatedPaymentStatusRef.current) {
+            updatePaymentStatus();
+            wasUpdatedPaymentStatusRef.current = true;
```

Comment:
Why do we have this ref here? I am guessing it is connected with strict mode double rendering...?

Reply:
Yes. There are some events which happen when the status is updated. I tried to rewrite the code so it can be run twice.